### PR TITLE
RD-631 Upgraded Cypress to v6

### PR DESCRIPTION
**Pull request by kubama**

### CODE REVIEW

File: test/cypress/integration/widgets/blueprint_catalog_spec.js [hunk 1]
```diff
@@ -33,9 +32,9 @@ describe('Blueprints catalog widget', () => {
             "Invalid blueprint - Plugin cloudify-aws-plugin (query: {'package_name': 'cloudify-aws-plugin'}) not found"
         );
 
-        cy.route('PUT', RegExp(`/console/sp\\?su=/blueprints/${blueprintName}`), {});
```
Comment (Gelio) [inline test/cypress/integration/widgets/blueprint_catalog_spec.js:36]: The URL is different. I assume it still matches fine, right?

File: test/cypress/integration/widgets/blueprints_spec.js [hunk 2]
```diff
@@ -79,7 +78,8 @@ describe('Blueprints widget', () => {
 
     describe('should render blueprint items', () => {
         beforeEach(() => {
-            cy.route(RegExp(`/console/sp\\?su=/blueprints`), 'fixture:blueprints/blueprints');
+            cy.interceptSp('GET', /blueprints.*&state=uploaded/).as('filteredBlueprints');
```
Comment (Gelio) [inline test/cypress/integration/widgets/blueprints_spec.js:81]: Would it not be cleaner to use the `{ query: { state: 'uploaded' } }` API?

File: test/cypress/integration/widgets/events_spec.js [hunk 3]
```diff
@@ -2,10 +2,7 @@ describe('Events/logs widget', () => {
     it('should show error cause', () => {
         cy.activate()
             .usePageMock('events', { fieldsToShow: ['Message', 'Workflow'], pageSize: 15 })
-            .route({
-                url: '/console/sp?su=/events?_size=15&_offset=0',
-                response: 'fixture:events/events.json'
-            })
+            .interceptSp('GET', '/events?_size=15&_offset=0', { fixture: 'events/events.json' })
```
Comment (Gelio) [inline test/cypress/integration/widgets/events_spec.js:5]: In order not to rely on the order of query parameters - how about using the `query: { } ` API?

File: test/cypress/integration/widgets/sites_spec.js [hunk 4]
```diff
@@ -184,13 +184,13 @@ describe('Sites Management', () => {
 
         cy.get('form :nth-child(2) > .field > .ui > input').should(
             'have.value',
-            '32.10118973232094, 34.80468750000001'
+            '32.10118973232094, 33.92578125000001'
```
Comment (Gelio) [inline test/cypress/integration/widgets/sites_spec.js:187]: :thinking: what's the reason for this change?

File: test/cypress/integration/widgets/spire_manager_spec.js [hunk 5]
```diff
@@ -11,49 +11,28 @@ describe('Spire Manager widget', () => {
...
-            method: 'POST',
-            url: '/console/sp?su=/executions',
-            response: {}
-        }).as('postExecutions');
+        cy.intercept('GET', '/console/wb/get_spire_deployments', {
+            fixture: `spire_deployments/3_deployments.json`
+        }).as('getSpireDeployments');
+        cy.intercept('GET', '/console/wb/get_cluster_status?deploymentId=rome', {
+            fixture: 'cluster_status/ok.json',
+            delayMs: 500
```
Comment (Gelio) [inline test/cypress/integration/widgets/spire_manager_spec.js:19]: If you decide to upgrade to 6.4.0, looks like the name was reverted to `delay`

https://docs.cypress.io/api/commands/intercept.html#History

File: test/cypress/support/commands.js [hunk 6]
```diff
@@ -36,7 +36,7 @@ Cypress.Commands.add('restoreState', () => cy.restoreLocalStorage());
 
 Cypress.Commands.add('waitUntilPageLoaded', () => {
     cy.log('Wait for widgets loaders to disappear');
-    cy.get('div.loader', { timeout: 10000 }).should('not.be.visible');
+    cy.get('div.loader:visible', { timeout: 10000 }).should('not.exist');
```
Comment (Gelio) [inline test/cypress/support/commands.js:39]: I'm having trouble understanding what is the `:visible` suffix in the selector :thinking: Can you help me out?

**Review by Gelio — COMMENTED** [review]
I like the local refactors you did while migrating :+1: The code looks cleaner now

File: test/cypress/integration/change_password_spec.js [hunk 7]
```diff
@@ -65,32 +65,25 @@ describe('Change Password modal', () => {
...
     });
 
     it('should not be available when LDAP is enabled', () => {
-        cy.server();
-        cy.route({
-            method: 'GET',
-            url: `/console/sp?su=/ldap`,
-            response: 'enabled'
-        }).as('ldap');
+        cy.interceptSp('GET', `/ldap`, 'enabled').as('ldap');
```
[the remark this answers is not in the document]
Comment (Gelio) [reply test/cypress/integration/change_password_spec.js:82]: Nevermind. I see now that it's a custom command :slightly_smiling_face:

File: test/cypress/integration/widgets/blueprint_catalog_spec.js (hunk 1 above)
Comment (Gelio) [reply test/cypress/integration/widgets/blueprint_catalog_spec.js:36]: Nevermind, I see it's the same thanks to `interceptSp`

File: test/cypress/integration/edit_mode_spec.js [hunk 8]
```diff
@@ -257,14 +256,14 @@ describe('Edit mode', () => {
...
-                    expect(xhr.status).equals(200);
+                cy.wait(service).then(({ request, response }) => {
+                    expect(request.headers).contain({ 'widget-id': widgetId });
+                    expect(response.statusCode).equals(200);
                 });
             const verifyResponse = text => cy.get('.widgetContent .ui.segment pre code').should('contain.text', text);
 
-            cy.route('GET', '/console/wb/manager?endpoint=version').as('managerService');
-            cy.route('GET', `/console/wb/request?url=${url}`).as('requestService');
+            cy.intercept('GET', '/console/wb/manager?endpoint=version').as('managerService');
```
Comment (qooban) [inline test/cypress/integration/edit_mode_spec.js:265]: Here probably you can use:
```suggestion
            cy.intercept({ method: 'GET', pathname: '/console/wb/manager', query: { endpoint: 'version' }}).as('managerService');
```

File: test/cypress/integration/widgets/deployments_spec.js [hunk 9]
```diff
@@ -175,14 +171,13 @@ describe('Deployments widget', () => {
     it('should allow to remove deployment', () => {
         const testDeploymentName = `${deploymentName}_remove`;
         cy.deployBlueprint(blueprintName, testDeploymentName);
-        cy.server();
-        cy.route('DELETE', `**/deployments/${testDeploymentName}?force=true`).as('deleteDeployment');
+        cy.interceptSp('DELETE', `/deployments/${testDeploymentName}?force=true`).as('deleteDeployment');
```
Comment (qooban) [inline test/cypress/integration/widgets/deployments_spec.js:174]: Probably you can extract query param from it and pass it in separate argument.

File: test/cypress/integration/widgets/sites_spec.js [hunk 10]
```diff
@@ -184,13 +184,13 @@ describe('Sites Management', () => {
...
         );
 
         // Click update
         cy.get('.actions > .green').click();
 
-        verifySiteRow(1, { name: siteWithLocation.name, location: '32.1011897323209, 34.8046875' });
+        verifySiteRow(1, { name: siteWithLocation.name, location: '32.1011897323209, 33.92578125' });
```
Comment (qooban) [inline test/cypress/integration/widgets/sites_spec.js:193]: Similar comment to the previous one. I suspect it is due to changes in Electron version after Cypress upgrade?

**Review by qooban — COMMENTED** [review]
Good that we are moving towards consistency 👍🏻

File: test/cypress/integration/widgets/blueprints_spec.js (hunk 2 above)
Comment (kubama) [reply test/cypress/integration/widgets/blueprints_spec.js:81]: Not possible here - see `interceptSp` implementation - the passed argument is already matched as a query parameter.
To be honest - I don't like `/console/sp` endpoint API - it uses `su` query parameter as manager URL which is hard to work with - since it is URL-encoded it's hard to read when it appears in any kind of logs. It also complicates usage of standard programming APIs, as it is in this case. In my opinion it should be refactored so the manager URL was part of the path, like this: `/console/sp/<manager_url>`. But this is for sure out of scope of this ticket.

File: test/cypress/integration/widgets/events_spec.js (hunk 3 above)
Comment (kubama) [reply test/cypress/integration/widgets/events_spec.js:5]: See my previous comment

File: test/cypress/integration/widgets/sites_spec.js (hunk 4 above)
Comment (kubama) [reply test/cypress/integration/widgets/sites_spec.js:187]: The test stopped passing after Cypress upgrade - I'm not 100% sure why as I didn't investigate it but I think it might be due to underlying Electron version bump. In fact, we were updating these values in this test in the past - you may check in git history if you are interested.

File: test/cypress/support/commands.js (hunk 6 above)
Comment (kubama) [reply test/cypress/support/commands.js:39]: https://api.jquery.com/visible-selector/

File: test/cypress/integration/auth_spec.js [hunk 11]
```diff
@@ -2,7 +2,7 @@ describe('Authentication', () => {
     beforeEach(() => cy.clearCookie('XSRF-TOKEN'));
 
     it('fails when token is not set in cookies', () => {
-        cy.visit('/console').waitUntilLoaded();
```
[the remark this answers is not in the document]
Comment (kubama) [reply test/cypress/integration/auth_spec.js:5]: It was waiting for widget loaders to complete. But in this it was performed on login page, so it was pointless.

File: package.json [hunk 12]
```diff
@@ -112,7 +112,7 @@
     "compression-webpack-plugin": "^6.1.1",
     "copy-webpack-plugin": "^6.3.2",
     "css-loader": "^3.2.0",
-    "cypress": "^5.1.0",
+    "cypress": "^6.3.0",
```
[the remark this answers is not in the document]
Comment (kubama) [reply package.json]: Upgraded

File: test/cypress/integration/widgets/spire_manager_spec.js (hunk 5 above)
Comment (kubama) [reply test/cypress/integration/widgets/spire_manager_spec.js]: Thanks for pointing this out

File: test/cypress/integration/widgets/deployments_spec.js (hunk 9 above)
Comment (kubama) [reply test/cypress/integration/widgets/deployments_spec.js:174]: Please see https://github.com/cloudify-cosmo/cloudify-stage/pull/1185#discussion_r568455830

File: test/cypress/integration/widgets/sites_spec.js (hunk 10 above)
Comment (kubama) [reply test/cypress/integration/widgets/sites_spec.js:193]: I didn't investigated it, but I think it's the most probable reason

File: test/cypress/integration/widgets/blueprints_spec.js (hunk 2 above)
Comment (qooban) [reply test/cypress/integration/widgets/blueprints_spec.js:81]: I like that idea (`/console/sp/<manager_url>`). Could you create a ticket for that in UI Technical Debt epic?

Comment (kubama) [reply test/cypress/integration/widgets/blueprints_spec.js:81]: Created RD-1277

File: test/cypress/integration/auth_spec.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (qooban) [reply test/cypress/integration/auth_spec.js:5]: Hmm... Are you sure it is not needed here? 

`waitUntilPageLoaded` wait for widgets loaders (if any present) to disappear.
`waitUntilLoaded` waits first for splash screen loader to disappear and then calls `waitUntilPageLoaded`. 

So AFAIR here it was used only for the first purpose - splash screen loader to disappear.
If we want to be very precise, probably we can extract "waiting for splash screen loader to disappear" functionality to another function and use it here.

[the remark this answers is not in the document]
Comment (kubama) [reply test/cypress/integration/auth_spec.js:5]: It's still not required in this particular test - we are not doing any assertions on the page content, only waiting to be redirected to `/login`

**Review by qooban — APPROVED** [review]

File: test/cypress/integration/auth_spec.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (Gelio) [reply test/cypress/integration/auth_spec.js:5]: Thanks for the explanation :slightly_smiling_face:

File: test/cypress/integration/widgets/blueprints_spec.js (hunk 2 above)
Comment (Gelio) [reply test/cypress/integration/widgets/blueprints_spec.js:81]: Great idea, guys :smile: :tada:

File: test/cypress/support/commands.js (hunk 6 above)
Comment (Gelio) [reply test/cypress/support/commands.js:39]: Good to know :slightly_smiling_face: thanks

**Review by Gelio — APPROVED** [review]
